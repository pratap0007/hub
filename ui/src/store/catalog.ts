import { types, Instance } from "mobx-state-tree";

export const Catalog = types
  .model({
    name: types.string,
    selected: false
  })
  .actions(self => ({
    toggle() {
      self.selected = !self.selected;
    }
  }));

export type ICatalog = Instance<typeof Catalog>;

export const CatalogStore = types
  .model("CatalogStore", {
    catalogList: types.array(Catalog)
  })

  .views(self => ({
    get count() {
      return self.catalogList.length;
    },
    get list() {
      const { catalogList } = self;
      return catalogList;
    },
    get selectedCatalogType() {
      const { catalogList } = self;
      return catalogList
        .filter((type: ICatalog) => type.selected)
        .map((type: ICatalog) => type.name);
    }
  }))

  .actions(self => ({
    add(item: any) {
      self.catalogList.push(item);
    },
    toggleCatalogType(kindName: string) {
      self.catalogList.forEach((kind: ICatalog) => {
        if (kind.name === kindName) {
          kind.toggle();
          return;
        }
      });
    },
    clearAll() {
      self.catalogList.map((item: ICatalog) => (item.selected = false));
    }
  }));
