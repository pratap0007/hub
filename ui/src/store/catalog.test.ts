import { CatalogStore, Catalog } from "./catalog";

describe("CatalogType", () => {
  it("can create a store and add a type", done => {
    const store = Catalog.create({
      name: "abc",
      selected: false
    });

    expect(store.name).toBe("abc");

    done();
  });

  it("it can toggle a type", done => {
    const store = CatalogStore.create();
    store.toggleCatalogType("official");
    expect(store.catalogList[0].selected).toBe(true);

    done();
  });
});
