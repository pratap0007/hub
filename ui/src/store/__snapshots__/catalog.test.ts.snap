// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store Object creates a catalog store 1`] = `
Object {
  "err": "",
  "isLoading": false,
  "items": Object {
    "1": Object {
      "id": 1,
      "name": "tekton",
      "selected": false,
      "type": "community",
    },
    "2": Object {
      "id": 2,
      "name": "openshift",
      "selected": false,
      "type": "community",
    },
  },
}
`;
